src/vs/server/node/server.main.ts

```diff
@@ -1014,12 +1014,20 @@ export async function main(options: IServerOptions): Promise<void> {
 		} else if (typeof options.port === 'number') {
 			port = options.port;
 		}
-		const host = parsedArgs.host || '0.0.0.0';
-		server.listen(port, host, () => {
-			const addressInfo = server.address() as net.AddressInfo;
-			const address = addressInfo.address === '0.0.0.0' || addressInfo.address === '127.0.0.1' ? 'localhost' : addressInfo.address;
-			const port = addressInfo.port === 80 ? '' : String(addressInfo.port);
-			logService.info(`Web UI available at http://${address}:${port}`);
-		});
+		function startServer(port: number) {
+			const host = parsedArgs.host || '0.0.0.0';
+			server.on('error', () => {
+				logService.warn(`Port ${port} is taken`);
```

Comment:
I think logging warning for all ports will not be ideal, Instead we can show after finding the port 🙇🏻 

Example
```
Port 8000 is not available, instead serving from 8004
```